### ISSUE TITLE
Transferring to uchardet organization?

### ISSUE THREAD

**Issue by Jehan** [BYVoid/uchardet#22]
Hi @BYVoid,

Would you mind if we create a "uchardet" organization in github, and transfer the uchardet project there? It will make it more "official" and stand up in the middle of the various forks.

As an alternative, we could be hosted by a friend organization, like on GNOME repository (i.e. the main repository will be out of github, though GNOME also has mirror of all its repositories here: https://github.com/GNOME). I have not asked the GNOME foundation yet, but I think they would accept. This would not make it a GNOME project, simply a friend project and still keeps independence.
This second alternative would likely be my favorite. :-) But I'm fine if you are absolutely attached to keep the main repository on github.

Thanks!

Jehan

**Comment by Jehan** [comment]
@BYVoid: I have sent an email to GNOME today to ask if it were possible to be hosted as a friend-of-GNOME project. I'll see what they say but that should not be a problem since uchardet is going to be used by some GNOME projects as well.
So if you want to voice your opinion and don't want the project to be moved there, do not hesitate to say it. :-) I don't want to give the impression I am "stealing" the project away from you or anything. I just feel like uchardet deserves a more stable repository than as a project on a personal github account.
In any case, you will be always welcome as co-maintainer wherever we are hosted if you wish to continue to code on uchardet. :-)

**Comment by Jehan** [comment]
I had an answer from GNOME and I got the authorization to host uchardet repository there.
So I will move the official upstream there in the coming days or weeks. If someone has a comment or a worry about it, please speak now.
@BYVoid in particular, if that is a problem for you, just tell me. :-)

**Comment by BYVoid** [comment]
I don't mind doing so. You can move it there.

Thank you very much for the maintenance.

**Comment by Jehan** [comment]
Thanks. For info, uchardet is now https://git.gnome.org/browse/uchardet/
And we have a small page in: https://wiki.gnome.org/Projects/uchardet
I'm only waiting for the project to make it in GNOME bugzilla (hopefully soon) before making the transfer official.

**Comment by BYVoid** [comment]
Good to hear that. Thanks for your work.

**Comment by Coacher** [comment]
Once the transfer is complete the official places for git repository, bug reports, and release tarballs will be at GNOME infrastructure?
What will be the status of this repository by then?
ATM by clicking on the very first `uchardet` link at https://wiki.gnome.org/Projects/uchardet I am redirected back to https://github.com/BYVoid/uchardet.

**Comment by Jehan** [comment]
> Once the transfer is complete the official places for git repository, bug reports, and release tarballs will be at GNOME infrastructure?

Yes. Just to make things clear though: it does not mean uchardet will start having high level dependencies to GNOME or its technology. GNOME is a friend project in which I contribute, and as such they will host us and provide infrastructure. But that's it.
In particular I plan on keeping it as small and low print as possible. No reason to change what is working. The point is only to give it more sustainability, and some visibility since I think this library is worth being used more massively (all other FOSS which I tested — I may have missed some alternatives — doing similar things are not performing that well). It is already used in [mpv](https://mpv.io/) (for subtitle encoding detection), and it is planned to be integrated into [gtksourceview](https://git.gnome.org/browse/gtksourceview/log/?h=wip/uchardet-2), which means [gedit](https://wiki.gnome.org/Apps/Gedit) and more (but that's a work in progress for when I'll have the time to finish). VLC also expressed their interest if someone were to provide a patch.
So I think this library should not be hosted under a single name but as a bigger project of its own. If I ever step down from maintenance myself, this will be much more easier to give it to someone interested.

>  What will be the status of this repository by then?

I won't delete code inside it (this is not my place to do so. The repository belongs to @BYVoid) but will add a note at the top of the README telling this is not the home of the project anymore and giving a link.

> ATM by clicking on the very first uchardet link

I know, but thanks for reminding me. This github repository is still the official place until uchardet gets into GNOME bug tracker so I kept the link on purpose for now. No need to move for somewhere else which does not propose the same basic features.

**Comment by Coacher** [comment]
Thank you very much for the detailed explanation!

**Comment by Jehan** [comment]
For info, GNOME is not responding to the bug report. I have no doubt they would accept, but this is not a priority to accept new projects, I guess.
In the meantime I thought that actually FreeDesktop may be a much more proper hosting organization. Uchardet is really not-desktop specific and a shared technology about interoperability (not between machines, but user-machine interoperability).

So I've also created a request to FreeDesktop: https://bugs.freedesktop.org/show_bug.cgi?id=95256
This may actually be a better fit.

**Comment by Coacher** [comment]
Hello again.

I can see at https://bugs.freedesktop.org/show_bug.cgi?id=95256 that fdo admins are more responsive and everything is almost ready for the transfer. So will it be fdo then or GNOME?

**Comment by Jehan** [comment]
It will be fdo, which anyway is a better fit for the project.
uchardet is more of a desktop-agnostic technology, made to improve any user experience. Well GNOME would have been OK as well. Many desktop-agnostic technology were born there. But it is still better to play as generic and open as possible from the start and avoid any useless desktop war which may occur. :-)

**Comment by Jehan** [comment]
We got everything from Freedesktop: a [repo](https://cgit.freedesktop.org/uchardet/uchardet/), a [product in the bug tracker](https://bugs.freedesktop.org/enter_bug.cgi?product=uchardet), wiki and upload (for releases) access.
I'll take care of the wiki page, and will make a new release in the next coming days. Then a commit here to announce the main page is moving (still leaving the history).
When this will be done, I will consider the transfer done and will close this report.

**Comment by Coacher** [comment]
Great news. Thanks!

**Comment by Coacher** [comment]
Hello. Any updates on this one?

**Comment by Jehan** [comment]
Yes. This is done. I have also made a new release 0.0.6 to officialize the move.
https://www.freedesktop.org/wiki/Software/uchardet/

I have updated the current github repository with a message atop the README to redirect people to the freedesktop hosting.
I will also make a blog post in a few days to spread the word a little.
Closing this!

**Comment by BYVoid** [comment]
Thanks for your great work.

**Comment by Coacher** [comment]
Great! Thank you very much!
